src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -347,13 +347,16 @@ public char[] ReadChars(int byteLen)
             return result;
         }
 
-        public void ReadBytes(byte[] output, int outputOffset, int len)
+        public void ReadBytes(Span<byte> output)
         {
-            Debug.Assert(len <= ReadBytesLeft);
-            System.Buffer.BlockCopy(Buffer, ReadPosition, output, outputOffset, len);
-            ReadPosition += len;
+            Debug.Assert(output.Length <= ReadBytesLeft);
+            new Span<byte>(Buffer, ReadPosition, output.Length).CopyTo(output);
+            ReadPosition += output.Length;
         }
 
+        public void ReadBytes(byte[] output, int outputOffset, int len)
```

Comment:
It makes to convert all other usages of `ReadBytes(byte[]..)` to `ReadBytes(Span<byte>)`, even when there isn't a huge benefit to it. But it can be done in another PR if you want.

Reply:
Yes, exactly :) And yeah, it's a different story, and we can also do that for char arrays. You want to do that in a separate PR?